### ISSUE TITLE
損失平均のサンプル重み付け

### ISSUE THREAD

**Issue by kurorosu** [kurorosu/pochitrain#138]
## 概要

- 損失平均がバッチ平均
- 最終バッチが小さい場合の偏り

### 1 サンプル数による平均

- `loss.item() * batch_size` の合算
- `total` での除算

## コード変更例

```python
# pochitrain/training/evaluator.py
total_loss += loss.item() * target.size(0)
```

## 対象ファイル

- `pochitrain/pochi_trainer.py` — 損失平均の算出変更
- `pochitrain/training/evaluator.py` — 損失平均の算出変更
- `tests/unit/test_core/test_pochi_trainer.py` — 平均計算のテスト更新
- `tests/unit/test_core/test_evaluator.py` — 平均計算のテスト更新

## Labels

`enhancement`

## Branch

`enhancement/sample-weighted-loss`
